agent/api/container/container_windows.go

```diff
@@ -15,8 +15,54 @@
 
 package container
 
+import (
+	"encoding/json"
+	"strings"
+
+	dockercontainer "github.com/docker/docker/api/types/container"
+	"github.com/pkg/errors"
+)
+
 const (
 	// DockerContainerMinimumMemoryInBytes is the minimum amount of
 	// memory to be allocated to a docker container
 	DockerContainerMinimumMemoryInBytes = 256 * 1024 * 1024 // 256MB
 )
+
+// RequiresCredentialSpec checks if container needs a credentialspec resource
+func (c *Container) RequiresCredentialSpec() bool {
+	credSpec, err := c.getCredentialSpec()
+	if err != nil || credSpec == "" {
+		return false
+	}
+
+	return true
+}
+
+// GetCredentialSpec is used to retrieve the current credentialspec resource
+func (c *Container) GetCredentialSpec() (string, error) {
+	return c.getCredentialSpec()
+}
+
+func (c *Container) getCredentialSpec() (string, error) {
```

Comment:
Do you know whether we have validation on backend side to check customers only specify one credential spec per container?

Reply:
Yes, the plan is to only support one `credentialspec` per container and the backend should have this check.